src/main/java/com/lpvs/entity/report/LPVSReportBuilder.java

```diff
@@ -0,0 +1,482 @@
+                                                                            Collectors
+                                                                                    .collectingAndThen(
+                                                                                            Collectors
+                                                                                                    .toList(),
+                                                                                            elements ->
+                                                                                                    new GroupInfo<>(
+                                                                                                            elements
+                                                                                                                    .size(),
+                                                                                                            elements))),
+                                                                    groupedByComponent ->
+                                                                            new GroupInfo<>(
+                                                                                    groupedByComponent
+                                                                                            .values()
+                                                                                            .stream()
+                                                                                            .mapToLong(
+                                                                                                    GroupInfo
+                                                                                                            ::getCount)
+                                                                                            .sum(),
+                                                                                    groupedByComponent))),
+                                                    groupedByLicense ->
+                                                            new GroupInfo<>(
+                                                                    groupedByLicense
+                                                                            .values()
+                                                                            .stream()
+                                                                            .mapToLong(
+                                                                                    GroupInfo
+                                                                                            ::getCount)
+                                                                            .sum(),
+                                                                    groupedByLicense))));
```

Comment:
to much for single instruction. Split required here.

How about:
```
Map<String, GroupInfo<?>> detectedLicenseInfo = filesScanResults.stream()
    .collect(Collectors.groupingBy(
        this::getFirstLicenseAccess,
        Collectors.collectingAndThen(
            Collectors.groupingBy(
                this::getFirstLicenseSpdxId,
                Collectors.collectingAndThen(
                    Collectors.groupingBy(
                        this::getComponentKey,
                        Collectors.collectingAndThen(
                            Collectors.toList(),
                            this::createGroupInfo
                        )
                    ),
                    this::summarizeGroupInfo
                )
            ),
            this::summarizeGroupInfo
        )
    ));
```
    
    and after that implement corresponding methods?
    
    together with created classes above it will be much easier to maintain.